src/internal/NeoEsp8266DmaMethod.h

```diff
@@ -330,15 +330,15 @@ template<typename T_SPEED> class NeoEsp8266DmaMethodBase
             _i2sBufDesc[indexDesc].sub_sof = 0;
             _i2sBufDesc[indexDesc].datalen = sizeof(_i2sZeroes);
             _i2sBufDesc[indexDesc].blocksize = sizeof(_i2sZeroes);
-            _i2sBufDesc[indexDesc].buf_ptr = reinterpret_cast<uint32_t>(_i2sZeroes);
+            _i2sBufDesc[indexDesc].buf_ptr = _i2sZeroes;
             _i2sBufDesc[indexDesc].unused = 0;
-            _i2sBufDesc[indexDesc].next_link_ptr = reinterpret_cast<uint32_t>(&(_i2sBufDesc[indexDesc + 1]));
+            _i2sBufDesc[indexDesc].next_link_ptr = &(_i2sBufDesc[indexDesc + 1]);
         }
 
         // the first state block will trigger the interrupt
         _i2sBufDesc[indexDesc - 2].eof = 1;
         // the last state block will loop to the first state block by defualt
-        _i2sBufDesc[indexDesc - 1].next_link_ptr = reinterpret_cast<uint32_t>(&(_i2sBufDesc[indexDesc - 2]));
+        _i2sBufDesc[indexDesc - 1].next_link_ptr = &(_i2sBufDesc[indexDesc - 2]);
```

Comment:
Why isn't this causing you a warning, casting between *uint32_t and *slc_queue_item?

Reply:
Indeed, this even leads to a compile error :see_no_evil: 

I realized I build tested this only for ESP32, that is why I haven't seen it.